ocaml/xapi/xapi_pool_helpers.ml

```diff
@@ -269,3 +269,221 @@ let apply_guest_agent_config ~__context =
         (Printexc.to_string e)
   in
   call_fn_on_slaves_then_master ~__context f
+
+module PeriodicUpdateSync = struct
+  let periodic_update_sync_task_name = "Periodic update synchronization"
+
+  exception UpdateSync_RetryNumExceeded of int
+
+  let seconds_random_within_a_day () =
```

Comment:
Would suggest to limit the randomness between from hour 01 to hour 23 to mitigate the possibility of two too close schedule, like now is 23:59, and the next schedule is 00:03 of tomorrow.

Reply:
Changed in another way: set the minimum interval between 2 run to 2 hours.